diff --git a/rtl/platform.vhd b/rtl/platform.vhd
--- a/rtl/platform.vhd
+++ b/rtl/platform.vhd
@@ -428,7 +428,7 @@
 	romb2_cs <= '1' when dn_addr(15 downto 12) = "1001" else '0';
 	romb3_cs <= '1' when dn_addr(15 downto 12) = "1010" else '0';
 
-	rom_inst : work.dpram generic map (14,8)
+	rom_inst : entity work.dpram generic map (14,8)
 	port map
 	(
 		clock_a   => clk_sys,
@@ -441,7 +441,7 @@
 		q_b       => rom_d_o
 	);
 
-	char1_rom_inst : work.dpram generic map (12,8)
+	char1_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a   => clk_sys,
@@ -454,7 +454,7 @@
 		q_b       => tilemap_o(1).tile_d(15 downto 8)
 	);
 
-	char2_rom_inst : work.dpram generic map (12,8)
+	char2_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a   => clk_sys,
@@ -470,7 +470,7 @@
 	spr_addr <= dn_addr(11 downto 0) when rst_sys = '1' else (sprite_i.a(11 downto 5) & '0' & sprite_i.a(3 downto 0));
 	spr_clk  <= clk_sys when rst_sys = '1' else clk_video;
 
-	sprite1_rom_inst : work.dpram generic map (12,8)
+	sprite1_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a	 => spr_clk,
@@ -486,7 +486,7 @@
 		q_b                     => sprite_o.d(23 downto 16)
 	);
 
-	sprite2_rom_inst : work.dpram generic map (12,8)
+	sprite2_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a	 => spr_clk,
@@ -504,7 +504,7 @@
 
    sprite_o.d(sprite_o.d'left downto 32) <= (others => '0');
 
-	bg1_rom_inst : work.dpram generic map (12,8)
+	bg1_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a   => clk_sys,
@@ -517,7 +517,7 @@
 		q_b       => bitmap_o(1).d(7 downto 0)  
 	);
 
-	bg2_rom_inst : work.dpram generic map (12,8)
+	bg2_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a   => clk_sys,
@@ -530,7 +530,7 @@
 		q_b       => bitmap_o(2).d(7 downto 0)  
 	);
 
-	bg3_rom_inst : work.dpram generic map (12,8)
+	bg3_rom_inst : entity work.dpram generic map (12,8)
 	port map
 	(
 		clock_a   => clk_sys,
diff --git a/rtl/moon_patrol_sound_board.vhd b/rtl/moon_patrol_sound_board.vhd
--- a/rtl/moon_patrol_sound_board.vhd
+++ b/rtl/moon_patrol_sound_board.vhd
@@ -363,7 +363,7 @@
 roms_cs <= '1' when dn_addr(15 downto 12) = "1011" else '0';
 
 -- cpu program rom
-cpu_prog_rom : work.dpram generic map (12,8)
+cpu_prog_rom : entity work.dpram generic map (12,8)
 port map
 (
 	clock_a   => clock_30,
@@ -377,7 +377,7 @@
 );
 
 -- cpu wram 
-cpu_ram : work.dpram generic map (7,8)
+cpu_ram : entity work.dpram generic map (7,8)
 port map
 (
 	clock_a   => clock_div(0),  -- 3p58/2,
